cmd/inspect.go

```diff
@@ -27,80 +27,123 @@ import (
 	"os"
 
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 
+	"go.k6.io/k6/core/local"
 	"go.k6.io/k6/js"
 	"go.k6.io/k6/lib"
 	"go.k6.io/k6/lib/metrics"
-	"go.k6.io/k6/loader"
+	"go.k6.io/k6/lib/types"
 )
 
-func getInspectCmd(logger logrus.FieldLogger) *cobra.Command {
+func getInspectCmd(logger *logrus.Logger) *cobra.Command {
+	var addExecReqs bool
+
 	// inspectCmd represents the inspect command
 	inspectCmd := &cobra.Command{
 		Use:   "inspect [file]",
 		Short: "Inspect a script or archive",
 		Long:  `Inspect a script or archive.`,
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			pwd, err := os.Getwd()
-			if err != nil {
-				return err
-			}
-			filesystems := loader.CreateFilesystems()
-			src, err := loader.ReadSource(logger, args[0], pwd, filesystems, os.Stdin)
+			src, filesystems, err := readSource(args[0], logger)
 			if err != nil {
 				return err
 			}
 
-			typ := runType
-			if typ == "" {
-				typ = detectType(src.Data)
-			}
-
 			runtimeOptions, err := getRuntimeOptions(cmd.Flags(), buildEnvMap(os.Environ()))
 			if err != nil {
 				return err
 			}
 			registry := metrics.NewRegistry()
-			_ = metrics.RegisterBuiltinMetrics(registry)
+			builtinMetrics := metrics.RegisterBuiltinMetrics(registry)
 
-			var (
-				opts lib.Options
-				b    *js.Bundle
-			)
-			switch typ {
+			var b *js.Bundle
+			switch getRunType(src) {
+			// this is an exhaustive list
 			case typeArchive:
 				var arc *lib.Archive
 				arc, err = lib.ReadArchive(bytes.NewBuffer(src.Data))
 				if err != nil {
 					return err
 				}
 				b, err = js.NewBundleFromArchive(logger, arc, runtimeOptions, registry)
-				if err != nil {
-					return err
-				}
-				opts = b.Options
+
 			case typeJS:
 				b, err = js.NewBundle(logger, src, filesystems, runtimeOptions, registry)
+			}
+			if err != nil {
+				return err
+			}
+
+			// ATM, output can take 2 forms: standard (equal to lib.Options struct) and extended, with additional fields.
+			inspectOutput := interface{}(b.Options)
+
+			if addExecReqs {
+				inspectOutput, err = addExecRequirements(b, builtinMetrics, registry, logger)
```

Comment:
Maybe I'm missing something but could we reuse the `newRunner` function from `cmd/run.go` and inject the runner here? 

https://github.com/grafana/k6/blob/5c14dae129a5eb6ed309c94a194370878b7b08e5/cmd/run.go#L390